rotkehlchen/chain/evm/decoding/curve_lend/decoder.py

```diff
@@ -617,14 +632,102 @@ def _decode_vault_controller_events(self, context: DecoderContext) -> DecodingOu
+                    event.amount == amount and
+                    event.address == context.tx_log.address
+            ):
+                event.counterparty = CPT_CURVE
+                event.product = EvmProduct.GAUGE
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                event.notes = f'Deposit {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} into {gauge_asset.symbol}'  # noqa: E501
+
+                paired_events_data = ([event], True)
+                from_event_type = HistoryEventType.RECEIVE
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.RECEIVE
+                to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                to_notes = f'Receive {amount} {gauge_asset.symbol} after depositing in curve lending vault gauge'  # noqa: E501
+
+            elif (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == amount and
+                    event.address == context.tx_log.address
+            ):
+                event.counterparty = CPT_CURVE
+                event.product = EvmProduct.GAUGE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Withdraw {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from {gauge_asset.symbol}'  # noqa: E501
+
+                paired_events_data = ([event], False)
+                from_event_type = HistoryEventType.SPEND
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.SPEND
+                to_event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                to_notes = f'Return {amount} {gauge_asset.symbol} after withdrawing from curve lending vault gauge'  # noqa: E501
+
+        return DecodingOutput(action_items=[] if paired_events_data is None else [
+            ActionItem(
+                action='transform',
+                from_event_type=from_event_type,  # type: ignore[arg-type]  # it cannot be none if paired_events_data is present
+                from_event_subtype=from_event_subtype,  # type: ignore[arg-type] # it cannot be none if paired_events_data is present
+                asset=gauge_asset,
+                amount=amount,
+                to_counterparty=CPT_CURVE,
+                to_notes=to_notes,
+                paired_events_data=paired_events_data,
+                to_event_type=to_event_type,
+                to_event_subtype=to_event_subtype,
+            ),
+        ])
+
+    def _maybe_enrich_curve_lending_gauge_rewards(self, context: EnricherContext) -> TransferEnrichmentOutput:  # noqa: E501
+        """May raise:
+        - UnknownAsset
```

Comment:
this one executes for transfers right? Can you state in the docstring what kind of event you are targeting?